File: sort-algorithm/bubble-sort/bubble-sort.rs
fn bubble_short_asc(mut in_arr: [i32; 10]) -> [i32; 10] {
    for i in 0..9 {
        for j in i+1..10 {
            if in_arr[j] <= in_arr[i] {
                let temp: i32 = in_arr[i];
                in_arr[i] = in_arr[j];
                in_arr[j] = temp;
            }
        }
    }
    return in_arr;
}

fn bubble_short_desc(mut in_arr: [i32; 10]) -> [i32; 10] {
    for i in 0..9 {
        for j in i+1..10 {
            if in_arr[j] >= in_arr[i] {
                let temp: i32 = in_arr[i];
                in_arr[i] = in_arr[j];
                in_arr[j] = temp;
            }
        }
    }
    return in_arr;
}

fn main() {
    let rand_arr: [i32; 10] = [233302, 32, 23923, 3421, 2, 889, 3, 4839, 32123, 19998];
    print!("initial array : ");
    for el in rand_arr {
        print!("{} ", el);
    }
    println!();

    let asc_arr: [i32; 10] = bubble_short_asc(rand_arr);
    for el in asc_arr {
        print!("{} ", el);
    }
    println!();

    let desc_arr: [i32; 10] = bubble_short_desc(rand_arr);
    for el in desc_arr {
        print!("{} ", el);
    }
    println!();
}

File: sort-algorithm/bubble-sort/bubble-sort.pl
sub bubble_sort_asc {
    $arr = $_[0];
    @in_arr = @arr;
    $max_arr = $_[1];

    for ($i = 0; $i < $max_arr-1; $i++) {
        for($j = $i+1; $j < $max_arr; $j++) {
            if (@in_arr[$j] <= @in_arr[$i]) {
                $temp = @in_arr[$i];
                @in_arr[$i] = @in_arr[$j];
                @in_arr[$j] = $temp;
            }
        }
    }

    return (@in_arr);
}

sub bubble_sort_desc {
    $arr = $_[0];
    @in_arr = @arr;
    $max_arr = $_[1];

    for ($i = 0; $i < $max_arr-1; $i++) {
        for($j = $i+1; $j < $max_arr; $j++) {
            if (@in_arr[$j] > @in_arr[$i]) {
                $temp = @in_arr[$i];
                @in_arr[$i] = @in_arr[$j];
                @in_arr[$j] = $temp;
            }
        }
    }

    return (@in_arr);
}

@arr = (123, 3129, 32331, 319320, 2931, 3, 454, 223, 123, 129339);
$max_arr = 10;
print join(" ", @arr);
print "\n";

@desc_arr = bubble_sort_desc(@arr, $max_arr);
@asc_arr = bubble_sort_asc(@arr, $max_arr);

print join(" ", @desc_arr);
print "\n";
print join(" ", @asc_arr);
print "\n";
